sdks/python/apache_beam/runners/interactive/cache_manager.py

```diff
@@ -0,0 +1,188 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import glob
+import os
+import shutil
+import tempfile
+import urllib
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam.io import filesystems
+from apache_beam.transforms import combiners
+
+
+class CacheManager(object):
+  """Abstract class for caching PCollections.
+
+  A PCollection cache is identified by labels, which consist of a prefix (either
+  'full' or 'sample') and a cache_label which is a hash of the PCollection.
+  """
+
+  def exists(self, *labels):
+    """Returns if the PCollection cache exists."""
+    raise NotImplementedError
+
+  def is_latest_version(self, version, *labels):
+    """Returns if the given version number is the latest."""
+    return version == self._latest_version(*labels)
+
+  def _latest_version(self, *labels):
+    """Returns the latest version number of the PCollection cache."""
+    raise NotImplementedError
+
+  def read(self, *labels):
+    """Return the PCollection as a list as well as the version number.
```

Comment:
It would be helpful if you explicitly document the return value(s) later in this docstring.  (I assume this returns something like `version_number, contents_list`)